modules/dbnd/src/dbnd/_core/cli/cmd_execute.py

```diff
@@ -7,13 +7,51 @@
 logger = logging.getLogger(__name__)
 
 
+def get_dbnd_version():
+    import dbnd
+
+    return dbnd.__version__
+
+
+def get_python_version():
+    import sys
+
+    return sys.version.rsplit(" ")[0]  # string like '3.8.6'
+
+
+class DbndVersionsClashError(Exception):
+    pass
+
+
 @click.group()
 @click.option("--dbnd-run", required=True, type=click.Path())
 # option that disables the tracking store access.
 @click.option("--disable-tracking-api", is_flag=True, default=False)
+@click.option("--expected-dbnd-version", default=None)
+@click.option("--expected-python-version", default=None)
 @click.pass_context
-def execute(ctx, dbnd_run, disable_tracking_api):
+def execute(
+    ctx, dbnd_run, disable_tracking_api, expected_dbnd_version, expected_python_version
+):
     """Execute databand primitives"""
+    if expected_python_version and expected_dbnd_version:
+        spark_python_version = get_python_version()
+        spark_dbnd_version = get_dbnd_version()
+        if expected_python_version != spark_python_version:
+            raise DbndVersionsClashError(
+                "You submmited job using Python {} but the Spark cluster uses Python {}. To "
+                "assure execution consistency use the same version in both places.".format(
+                    expected_python_version, spark_python_version
+                )
+            )
+        if expected_dbnd_version != spark_dbnd_version:
+            raise DbndVersionsClashError(
```

Comment:
Why raise? I think we're better off warning the user, and if code works - then so be it. WDYT?

Reply:
My main worry is the following situation:
- user submits the job
- something crashes miserably
- the error is due to databand version mismatch
- user misses a warning
- user reaches to dbnd for support

If we know that we are backward / forward compatible within major version then we should only check if the major is the same. As I'm not that familiar with what users tend to do, I'm happy to do what you suggest as best approach.

Reply:
I think that we can't take this decision for the user. We do our best -- write warning, a lot of text so its noticeable, but keep trying to run (best effort). I would simply write this in the message.
"You submitted job using version ... To assure execution consistency use the same version in both places. Continuing execution in best effort...
Users will notice, we differentiate spark engine logs and our bootstrap logs, I think this is our best approach, so reduce to warning 💯